src/partitioner/local/RDFPartitioner.cpp

```diff
@@ -57,19 +61,128 @@ void RDFPartitioner::loadDataSet(string inputFilePath, string outputFilePath, in
-        std::cout << "subject_str---" << subject_str << std::endl;
-        std::cout << "predicate_str---" << predicate_str << std::endl;
-        std::cout << "object_str---" << object_str << std::endl;
 
 
+
+        long firstVertex = addToStore(&nodes, subject_str);
+
+
+        long relation = addToStore(&predicates, predicate_str);
+
+        long secondVertex = addToStore(&nodes, object_str);
+
+
+        std::string secondVertexStr = std::to_string(secondVertex);
+
+        addToMap(&relationsMap, firstVertex, relation, secondVertexStr);
+
+
+        std::set<long> vertexSet = graphStorage.find(firstVertex)->second;
+
+        for (auto it = vertexSet.begin(); it != vertexSet.end(); ++it)
+
+            if (vertexSet.empty()) {
+
+                vertexSet = set<long>();
+                vertexSet.insert(secondVertex);
+                edgeCount++;
+                graphStorage.insert({firstVertex, vertexSet});
+            } else if (!vertexSet.empty()) {
+
+                if (vertexSet.insert(secondVertex).second) {
+                    edgeCount++;
+                }
+            }
+
     }
+    writeRelationData();
 }
 
+long RDFPartitioner::addToStore(std::map<string, long> *map, string URI) {
+    long id;
+    id = map->size();
+
+    auto search = map->find(URI);
+    if (search != map->end()) {
+        return search->second;
+    } else {
+
+    }
+
+
+    if (*map == nodes) {
+        nodesTemp.insert({id, URI});
+
+
+    } else if (*map == predicates) {
+        predicatesTemp.insert({id, URI});
+
+    }
+    map->insert({URI, id});
+    return id;
+}
+
+
+void RDFPartitioner::addToMap(std::map<long, std::map<long, std::vector<string> >> *map, long vertex, long relation,
+                              string value) {
+    std::map<long, std::vector<string>> miniMap = map->find(vertex)->second;
+
+
+    if (!miniMap.empty()) {
+
+        std::vector<string> list = miniMap.find(0)->second;
+
+        if (!list.empty()) {
+
+            list.push_back(value);
+        } else if (list.empty()) {
+
+            list = std::vector<string>();
+            list.push_back(value);
+            miniMap.insert({relation, list});
+        }
+
+    } else if (miniMap.empty()) {
+
+        std::vector<string> list;
+        list.push_back(value);
+        miniMap = std::map<long, vector<string>>();
+        miniMap.insert({relation, list});
+        map->insert({vertex, miniMap});
+
+    }
+}
+
+
 void RDFPartitioner::convert(string graphName, int graphID, string inputFilePath, string outputFilePath, int nParts,
                              bool isDistributedCentralPartitions, int nThreads, int nPlaces) {
     convertWithoutDistribution(graphName, graphID, inputFilePath, outputFilePath, nParts,
                                isDistributedCentralPartitions, nThreads, nPlaces);
 
     distributeEdges();
 
+}
+
+void RDFPartitioner::writeRelationData() {
+    ofstream file;
+    file.open("RDFdata.txt");
```

Comment:
This file seems to be a temp file. It should not be created in the same location where the application binary resides. It should be created in some different location such as working directory or at /tmp.

Reply:
Ok I'll do it that way